### PR TITLE
jpm sign: show debug logging with --verbose

### PR DESCRIPTION
**Pull request by kumar303**
Normal output : 

````
Creating XPI
JPM [info] XPI created at /Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi (25ms)
Created XPI at /Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi
JPM [info] Created XPI for signing: /Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi
JPM [error] Server response: Version already exists. ( status: 409 )
JPM [info] FAIL
````

With `--verbose` : 

````
JPM [info] verbose set
JPM [info] Checking compatability bootstrap.js and install.rdf for xpi
Validating the manifest
JPM [info] Creating fallbacks if they are necessary..
Creating XPI
JPM [info] Creating XPI...
JPM [warning] .jpmignore does not exist, fallback to use default filter rules
JPM [info] Adding: /Users/kumar/tmp/addons/scratchX-3/README.md
JPM [info] Adding: /Users/kumar/tmp/addons/scratchX-3/bootstrap.js
JPM [info] Adding: /Users/kumar/tmp/addons/scratchX-3/index.js
JPM [info] Adding: /Users/kumar/tmp/addons/scratchX-3/install.rdf
JPM [info] Adding: /Users/kumar/tmp/addons/scratchX-3/package.json
JPM [info] Adding: /Users/kumar/tmp/addons/scratchX-3/scratch_x_3-0.0.1-fx+an.xpi
JPM [info] XPI created at /Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi (26ms)
JPM [info] Removing fallbacks if they were necessary..
JPM [info] Creating updateRDF...
Created XPI at /Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi
JPM [info] Created XPI for signing: /Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi
JPM [debug] [API] -> { url: 'http://olympia.dev/api/v3/addons/%40scratchx-3/versions/0.0.1/',
  formData:
   { upload:
      { _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: undefined,
        path: '/Users/kumar/tmp/addons/scratchX-3/@scratchx-3-0.0.1.xpi',
        fd: null,
        flags: 'r',
        mode: 438,
        start: undefined,
        end: undefined,
        autoClose: true,
        pos: undefined } },
  headers:
   { Authorization: 'JWT eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ1c2VyOjE6Nzc2IiwiaWF0IjoxNDQ5MDc2NDYyLCJleHAiOjE0NDkwNzY1MjJ9.UR9X6bH_WHnwgRDdpDEFwc5ZAbXpYXU3EeeOSzNvg0M',
     Accept: 'application/json' } }
JPM [debug] [API] <- { headers:
   { server: 'nginx/1.9.5',
     date: 'Wed, 02 Dec 2015 17:14:22 GMT',
     'content-type': 'application/json',
     'transfer-encoding': 'chunked',
     connection: 'close',
     'content-security-policy-report-only': 'script-src \'self\' https://www.google.com https://www.paypalobjects.com https://ssl.google-analytics.com; default-src * data:; style-src * \'unsafe-inline\'; frame-src https://ssl.google-analytics.com; object-src \'none\'; report-uri /services/csp/report',
     vary: 'Accept, X-Mobile, User-Agent',
     'x-frame-options': 'DENY',
     allow: 'GET, PUT, HEAD, OPTIONS',
     'set-cookie': [ 'multidb_pin_writes=y; expires=Wed, 02-Dec-2015 17:14:37 GMT; Max-Age=15; Path=/' ] },
  response: { error: 'Version already exists.' } }
JPM [error] Server response: Version already exists. ( status: 409 )
JPM [info] FAIL
````

### CODE REVIEW

File: lib/amo-client.js [hunk 1]
```diff
@@ -428,6 +434,17 @@ Client.prototype.request = function(method, requestConf, options) {
 
 
 /**
+ * Output some debugging info if this instance is configured for it.
+ */
+Client.prototype.debug = function() {
+  if (!this.debugLogging) {
+    return;
+  }
+  logger.debug.apply(logger, arguments);
```
Comment (mstriemer) [inline lib/amo-client.js]: Seems like this could move into the `if` if you remove the `!`.

**Comment by mstriemer** [comment]
I don't see any tests that check that the `debugLogging` setting actually does anything, that might be useful.

I feel like the JWT should be hidden from the log. This seems like the sort of thing that we might ask someone to pastebin so it should be okay the make the results public.

r+wc

**Comment by mstriemer** [comment]
Perhaps hide `cookie` and `set-cookie` as well in case we start setting session cookies by accident.

**Comment by kumar303** [comment]
Hmm, yeah, maybe I'll hide some obfuscate some values. The JWT is pretty useless (as long as we add nonce checks) but yeah I can obfuscate that.

**Pull request merged**